.github/workflows/sync-main.yml

```diff
@@ -0,0 +1,48 @@
+name: automatic sync main branch from llvm-project to llvm
+
+on:
+  schedule:
+  - cron: '/10 * * * *'
+jobs:
+  sync:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          # persist-credentials: false allows us to use our own credentials for
+          # pushing to the repository.  Otherwise, the default github actions token
+          # is used.
+          persist-credentials: false
+          fetch-depth: 0
```

Comment:
I just noticed that we fetch all tags and branches. @hanzhan1, is it really needed? 
I would expect that we need to fetch the HEAD of `main` branch only. Right?